src/lib.rs

```diff
@@ -37,6 +37,7 @@
     clippy::inline_always,
     clippy::missing_errors_doc,
     clippy::module_name_repetitions,
+    clippy::needless_range_loop,
```

Comment:
I put it in here because I stylistically disagree with the lint. The lint fires here:

https://github.com/david-a-wheeler/metamath-knife/blob/96e4ae0e68615ed9482354690609b8e61011131c/src/verify_markup.rs#L476-L480

and it proposes:
```
warning: the loop variable `i` is used to index `buf`
   --> src/verify_markup.rs:476:26
    |
476 |                 for i in sp.start as usize..sp.end as usize {
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:21:9
    |
21  | #![warn(clippy::all, clippy::pedantic, clippy::nursery, clippy::cargo)]
    |         ^^^^^^^^^^^
    = note: `#[warn(clippy::needless_range_loop)]` implied by `#[warn(clippy::all)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
help: consider using an iterator
    |
476 |                 for (i, <item>) in buf.iter().enumerate().take(sp.end as usize).skip(sp.start as usize) {
    |                     ~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The use of `take` and `skip` is significantly worse readability-wise than the original code. It's not primarily an optimization concern; in fact the suggestion is probably faster because it only does a loop termination check instead of a loop termination check and a bounds check on every iteration. (This unnecessary bounds check in the loop could be fixed in the plain for-loop version by putting `assert!(sp.end as usize <= buf.len())` before the loop.)